Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    /// <summary>
+    /// Raycasts for collision in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is something in the way, else false.</returns>
+    private bool IsCollisionBlockingMovement(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection,1f, collisionLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Raycasts for slow terrain in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is slow terrain in the way, else false.</returns>
+    private bool IsEnteringSlowTerrain(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection, 1f, slowTerrainLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Removes the less influential movement axis and normalizes what's left
+    /// </summary>
+    private void NormalizeMovementVector()
+    {
+        normalizedMovementDirection = movementInput;
+        if (Mathf.Abs(normalizedMovementDirection.x) >= Math.Abs(normalizedMovementDirection.y))
+            normalizedMovementDirection.y = 0;
+        else
+            normalizedMovementDirection.x = 0;
+        normalizedMovementDirection = normalizedMovementDirection.normalized;
+    }
+
+    /// <summary>
+    /// Checks normalized movement and returns a direction used to activate player animations
+    /// </summary>
+    /// <returns></returns>
+    private directions GetDirectionFromNormalizedMovement()
+    {
+        if (normalizedMovementDirection == Vector2.down)
+            return directions.Down;
+        if (normalizedMovementDirection == Vector2.up)
+            return directions.Up;
+        if (normalizedMovementDirection == Vector2.left)
+            return directions.Left;
+        else
+            return directions.Right;
+    }
+
+    /// <summary>
+    /// Move the player in a set interval aligned with the tilemap
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator MovementRoutine()
+    {
+        isMoving = true;
+        NormalizeMovementVector();
+        directions movementDirection = GetDirectionFromNormalizedMovement();
+        if (IsCollisionBlockingMovement(movementDirection))
+        {
+            // TODO Play bump audio
+            PlayAnim(movementDirection, false, true);
+        }
+        else
+        {
+            Vector3 targetPosition = transform.position + new Vector3(normalizedMovementDirection.x, normalizedMovementDirection.y);
+            float startTime = Time.time;
+            if (IsEnteringSlowTerrain(movementDirection) || isInSlowTerrain)
+            {
+                PlayAnim(movementDirection, true, false);
+                while (Vector3.Distance(targetPosition, transform.position) > movementMargin)
+                {
```

Comment:
Make a method called Move where you figure out the target destination for the current frame and move there. Don't worry about slow movement for now.

The reason to do this is that you are repeating logic within this coroutine (multiple lerp calls, etc). Plus, we can make the code more expressive by having methods do fewer, more specific things.